taxcalc/parameters.py

```diff
@@ -387,9 +380,6 @@ def _update(self, year_mods):
         for name in unused_names:
             used_names.add(name)
             pname = name[:-4]  # root parameter name
-            if pname not in self._vals:
-                msg = 'root parameter name {} not in values dictionary'
-                raise ValueError(msg.format(pname))
             pindexed = year_mods[year][name]
```

Comment:
Paths to this section of code:

`Policy.implement_reform` --> `ParametersBase._update`

`Policy._validate_param_names_values` is called before `ParametersBase._update`. `Policy._validate_param_names_values` has a similar check: 

```
pname = name[:-4]  # root parameter name
if pname not in param_names:
    msg = '{} {} unknown parameter name'
    self.reform_errors += (
        'ERROR: ' + msg.format(year, name) + '\n'
)
```